add "core concepts" getting started page (#14534)
home-assistant/home-assistant.github.io: source/getting-started/core-concepts.markdown

Entities can be grouped into **areas**, which roughly represent a room in your house. Common areas are “kitchen”, “office”, “basement” and “yard”.

You can reference multiple entities at one time by putting them into a **[group](/integrations/group)**

_Advanced users can write [scripts](/docs/scripts), special entities that run custom code when they’re triggered_

## Automation

**[Automations](/docs/automation)** let you go further by telling Home Assistant to automatically do things without you having to hit a button yourself. For example, you could write an automation that says “turn on the lights when it gets dark if I’m home”.

An automation has three main parts:

- **[Triggers](/docs/automation/trigger)** - something that starts the automation, like the sun going down or a motion sensor noticing movement
- **[Conditions](/docs/automation/condition)** - let you decide whether to complete the automation. For example: "only if someone is home".
- **[Action](/docs/automation/action)** - what the automation does, like "turn on lights"

**[Templates](/docs/automation/templating)** let advanced users use variables and logic in their automations

## State

> Home Assistant remembers the **[state](/docs/configuration/state_object)** of each entity. Think of state as in "what state is this in?" For example, a light could have two states: ‘on’ and ‘off’, and a garage door could be in one of two states: ‘open’ or ‘closed’.

Reviewer: This leads me to believe that mutliple states are stored at one time.  Possibly change
```
For example, a light could have two states: ‘on’ and ‘off’
```
to:
```
For example, a light could have one of two states: ‘on’ and ‘off’
```

Author: love it, will fix 👍

---
title: "Core Concepts"
description: "How Home Assistant works."
---

Home Assistant, like all home automation software, allows you to control smart devices in your home. This page will explain the parts of Home Assistant and how they relate.

## Devices

In Home Assistant, a **device** is a physical object in your connected home. This would be something like a smart lightbulb or a Chromecast.

Home Assistant has a **discovery** feature, which scans for new devices constantly and adds them to your Home Assistant automatically.

**Device trackers** are devices that track the position of something or someone, like a person or a cellphone.

## Entities

Because many devices do more than one thing, things in Home Assistant are controlled using **entities**. An entity is a single function or property of a device, like a single sensor, light, or switch.

A device can provide multiple entities. For example, a device called “Sam’s Phone” might have entities like location, battery level, and step count. Some devices like lightbulbs only have a single entity, and devices like hubs may have no entities at all.

> To get entities to appear in Home Assistant, you must define them in your [`configuration.yaml` file](/docs/configuration) in this format: `entity_id: domain.object_id`. The **domain** is a built-in category of entities that you manage in a similar way. You can get the object ID by ________?

Author: Could I have some help with this sentence?

Reviewer: line 20 is great.  Line 22 seems like it's out of place.  That format will only be used in specific spots in configuration.yaml.

Author: Thanks @Petro31 , definitely struggling with line 22. How's something like this?

```
To get entities to appear in Home Assistant, you must define them in your [`configuration.yaml` file].

The **domain** is a built-in category of entities that you manage in a similar way. For example, all lights are part of the "light" domain.
```

---
title: "Core Concepts"
description: "How Home Assistant works."
---

Home Assistant, like all home automation software, allows you to control smart devices in your home. This page will explain the parts of Home Assistant and how they relate.

## Devices

> In Home Assistant, a **device** is a physical object in your connected home. An example would be a lightbulb, a cell phone, or a smart hub.

Reviewer 2: Since you mention Cell phone below I would remove it here. Maybe replace with Chrome Cast or Amazon Echo

Author: agreed--done

> In Home Assistant, a **device** is a physical object in your connected home. An example would be a lightbulb, a Chromecast, or a smart hub.

Reviewer 3: A "smart hub" might be a bad example, as they don't have to be a device themself, however, can provide additional other devices.

> Home Assistant has a **discovery** feature, which scans for new devices as they’re added and adds them to your Home Assistant automatically.

Reviewer 3: Removing the passive voice:
```suggestion
Home Assistant has a **discovery** feature, which constantly scans for new devices and adds them to your Home Assistant automatically.
```

However, we don't add them automatically

> **Device trackers** are devices like cell phones that can move around.

Reviewer 3: Yes / no. Device trackers are devices that track the position of something or someone, for example, a person, cellphone.

## Entities

> Because many devices do more than one thing, most things in Home Assistant use  **entities**. An entity is a single function or property of a device, like a single sensor, light, or switch.

Reviewer 3: > most things in Home Assistant use

Most? What is the alternative?

> A device can provide multiple entities. For example, a device called “Sam’s Phone” might have entities like location, battery level, and step count. Some devices like lightbulbs only have a single entity, and devices like hubs may have no entities at all.

Reviewer 3: I think it is more clear to somewhere put up that devices can provide other devices or entities.

Author: If I added "A device can also provide other devices", would that be true and fixing the issue?

Reviewer 3: Yeah or: 

`A device can provide, one or more, entities or other devices it is connected to.`

Some entities are useful for controlling things in your home:

> - **Switches** let you check if something is switched on or off?

Reviewer 3: It is a question `?` ?

- **Sensors** can be a wide range of things that watch for changes in the environment, like motion sensors and temperature sensors
- **Binary sensors** detect something that is either on or off

> - **[Scripts](https://www.home-assistant.io/docs/scripts/)** are special entities that run custom code when they’re triggered

Reviewer 3: Don't use full URL's instead pick them up from the root, for example: `/docs/scripts`.

It is unclear to me why scripts is linked, and others are not.

Author: Fair--this happened because of the markdown converter I used 😳

I also added more links, but there were lots of terms I couldn't find links for. If anyone can add links to the non-linked bold text I'd appreciate the help.

Other entities are there to respond to Home Assistant:

- **Lights** can turn on and off, and sometimes dim and change color

> - **Covers** are things that can move up and down, like garage doors and smart blinds

Reviewer 3: Why are these here? And switch is listed above? Why is scripts (which is considered highly advanced) listed above and a basic common thing like "Lights" here?

Author: I noted that scripts are advanced, but I'd like to keep a mention of them here so that people coming across scripts in Forums understand how they fit in. Maybe later there can be an "advanced concepts" page that's for more experienced people, but not developers?

Where should switches go? I just don't know how it works so I'd love a suggestion 😄

> Some entities are part of a **domain**, or a built-in category of entities that are managed in a similar way. For example, there is a “lights” domain in Home Assistant.

Reviewer 4: **All** entities have a domain don't they?

`entity_id: domain.object_id`

Reviewer 3: Yep, entities are always in a domain.

Reviewer 3: also `are managed` is a passive voice, try making it active.

Author: fixed passive voice!

Good to know that all entities have a domain. What's an object id? I'll take a stab at clarifying but it will probably be wrong.

Reviewer 4: In the example entity id `switch.electric_boogaloo`  the domain is `switch` and the object id is `electric_boogaloo`

Reviewer 4: Not sure the object id is necessary to add, but do change "Some entities are part of a **domain**" to "All entities are part of a **domain**"

Author: fixed!

Entities can be grouped into **areas**, which roughly represent a room in your house. Common areas are “kitchen”, “office”, “basement” and “yard”.

You can reference multiple entities at one time by putting them into a **[group](https://www.home-assistant.io/integrations/group/)**

## Automation

> **[Automations](https://www.home-assistant.io/docs/automation/)** let you go father by telling Home Assistant to automatically do things without you having to hit a button yourself. For example, you could write an automation that says “turn on the lights when it gets dark if I’m home”.

Reviewer 4: " father"? Who's your daddy and why is he important to automation? Suggest "further".

Although they are often used interchangeably, "further" and "fa**r**ther" don't have exactly the same meaning. Basically, "farther" refers to actual distances between objects while further refers to figurative distances or something that is additional or more.

Author: Lol, fixed 🙂

An automation has three main parts:

- **[Triggers](https://www.home-assistant.io/docs/automation/trigger/)** - something that starts the automation, like the sun going down or a motion sensor noticing movement
- **[Conditions](https://www.home-assistant.io/docs/automation/condition/)** - let you decide whether to complete the automation. For example: "only if someone is home".
- **[Action](https://www.home-assistant.io/docs/automation/action/)** - what the automation does, like "turn on lights"

**[Templates](https://www.home-assistant.io/docs/automation/templating/)** let advanced users use variables and logic in their automations

## Events and state

> Entities in Home Assistant are controlled by sending **[events](https://www.home-assistant.io/docs/configuration/events/)**.

Reviewer 3: Technically, under the hood, this is sort of true. However, is this really the experience from the user end? Is calling a service an event? I think it is better to set is as an action. For example, a service is an action.

The reasoning is that from the user end, we actually also handle events (e.g., in device automations). And you really don't want to create confusion around that.

Author: This section feels super wrong to me, but I just don't know how to fix it. What's an action? What's a service? I'll do my best but will probably need help. I'm also confused 🙂

Author: I just removed this sentence entirely

> Home Assistant remembers the **[state](https://www.home-assistant.io/docs/configuration/state_object/)** of each entity. The state means what *mode* or *condition* the entity is in this moment. For example, a light could have two states: ‘on’ and ‘off’, and a garage door could be in one of two states: ‘open’ or ‘closed’.

Reviewer 3: mode & conditions are different terms and things in the Home Assistant world. Let's not put that up the same level as a state.

A state represents the actual physical state a device has in the real world.

Author: whoops!

I'll try a different way of defining it, don't want to use the word in its own definition if i can help it though

> A **[scene](https://www.home-assistant.io/docs/scene/)** is a list of entities along with the state you want them each to be in. When you trigger that scene, Home Assistant will figure out the right events to get each of those entities into that state.

Reviewer 3: ```suggestion
A **[scene](/docs/scene/)** is a list of entities along with the state you want them each to be in. When you trigger that scene, Home Assistant will ensure all the entities get into the state set in that scene.
```

> For example, let’s imagine you have a scene called “welcome home” that says “lights on, door unlocked”. If you turn on that scene, Home Assistant will check if your lights are on and turn them on if they’re off. Then it will check if your door is unlocked, and unlock it if it’s locked.

Reviewer 3: ```suggestion
For example, let’s imagine you have a “welcome home” scene that says “lights on, door unlocked”. If you turn on that scene, Home Assistant will check if your lights are on and turn them on if they’re off. Then it will check if your door is unlocked, and unlock it if it’s locked.
```

## More concepts

- **Zones** are geographical areas on earth. You can check if a device tracker is inside a zone as a trigger or condition in an automation. For example, “turn on the lights when Sam’s Phone comes into the ‘home’ zone”.

> ## Integrations

Author: "Formerly known as components?"

Reviewer 2: Integrations have always been integrations I think. We should move this to the top of the page above devices. Integrations created Devices. Devices create Entities (which I think we should change the name of eventually).

Author: done

Home Assistant is set up to work with devices and services from lots of different companies, including voice assistants, smart devices, cameras, and other home automation software. You can make home Assistant work with these services by installing **[integrations](https://www.home-assistant.io/integrations)**.

Once installed, integrations can make new entities available, add panels to Lovelace, and more.

> _Not to be confused with add-ons._

Author: Is there a link I should be using here?

Author: Also, what are platforms and packages? Are they relevant here?

> _Formerly known as components; not to be confused with add-ons._

Reviewer 3: If someone is new here, please don't mention components. It isn't used anymore, it isn't listed in the docs anymore.

Furthermore, calling out add-ons out of the blue without further text, reference or explanation is confusion. As it instantly pops the question in the reader's head: "So what is an add-on?" (which also is distracting).

Author: > If someone is new here, please don't mention components. It isn't used anymore, it isn't listed in the docs anymore.

In my experience as a newbie googling and such, the word has definitely come up. It might be clearer to explicitly state why so nobody things "this documentation is incomplete, what's a component...". I can remove if you feel very strongly though.

> Furthermore, calling out add-ons out of the blue without further text, reference or explanation is confusion. As it instantly pops the question in the reader's head: "So what is an add-on?" (which also is distracting).

Should I say more about add-ons, or remove it? I'm confused, add ons have come up in multiple tutorials and they're right there in the HA lovelace UI... are they "under the hood" or not?

Reviewer 3: Right now, it is listed as a loose term, so:

>  I'm confused, add ons have come up in multiple tutorials and they're right there in the HA lovelace UI... are they "under the hood" or not?

☝️  Your own statement right there in the response shows the confusion. You highlight the possible confusion in the text, but you are not solving / removing the confusion.

I think we should either clarify what they are and fix the possible confusion, or not mention it at all.

## The Home Assistant software

So now you know what Home Assistant is. But how do you start using it?

> You probably want to [install Home Assistant](https://www.home-assistant.io/getting-started/) using a disk image, which contains everything you need to run Home Assistant and keep it updated.

Reviewer 3: Wait? This page is part of the getting started? So, why link to it?

Author: If they got to "core concepts" from another place, they may need a link to the installation instructions, which live in the "getting started" root

Reviewer 3: So it isn't getting started tutorial then? I'm confused. Are you aiming for adding something to the getting started guide, or writing something new?

If you’re feeling adventurous, you can also install parts of Home Assistant separately:

- **Core** - the main "engine" of Home Assistant that handles reading data, controlling devices, and running automations

> - **Supervisor** makes sure everything is up to date, installs add-ons, runs backups, and more

Reviewer 3: You cannot install this separately and is out of scope for the getting started guide. The getting started guide is focussed around installing the Home Assistant OS installation method.

Author: moved to an "advanced" note (just in case they CTRL-F in core concepts trying to understand what Supervisor is, as I did)

> Learn more about [different installation methods here](https://community.home-assistant.io/t/home-assistant-installation-methods/207703).

Reviewer 3: We don't support community installation methods. Please link to official documentation sources

Author: removed

Most people control Home Assistant with its built-in frontend, **Lovelace**. Lovelace is the user interface in your browser and the mobile apps.

Lovelace is organized like this:

- **Panels** are top-level pages in Lovelace. Examples are your Dashboard, the Map, the History panel and the Configuration page.
- **[Dashboard](https://www.home-assistant.io/lovelace/dashboards-and-views)** panels are the main way to see and control your home. You can add multiple custom dashboards in addition to the one that comes by default.
- **[Views](https://www.home-assistant.io/lovelace/dashboards-and-views)** are just tabs inside of dashboards. You can easily toggle between views on the same dashboard. You can customize the name and an icon for each view.

> - **Cards** are smaller pieces of UI inside a view. You can generally see multiple cards at once. Home Assistant comes with built in cards, but you can also write your own.

Reviewer 3: Why is this at the bottom all of a sudden?

Author: I'm going back and forth on the flow. Where would you like it to live?

Reviewer 3: I dunno, I just feel like this is pasted in at the end and thus a bit out of place. (disconnected from the story).
